lib/firmenwissen/http_request.rb

```diff
@@ -22,12 +22,23 @@ def request
       @request ||= begin
         Net::HTTP::Get.new(uri).tap do |req|
           req.basic_auth(config.user, config.password)
+          req.add_field('Cookie', session_cookie) if config.persistent_session? && !session_cookie.empty?
         end
       end
     end
 
     def config
       @config ||= Firmenwissen.configuration.merge(options)
     end
+
+    def extract_session(response)
+      return unless config.persistent_session?
+
+      Firmenwissen::Session.update_from_set_cookie_headers(response.get_fields('Set-Cookie'))
+    end
+
+    def session_cookie
```

Comment:
Does that name refer to the session in the Firmenwissen API or to the (pseudo) session created here in this gem by storing the cookies?

Reply:
It's the name for the set of cookies that have been set by the API. `JSESSIONID` and `SERVERID_SEARCH` are the 2 cookies that prompted this change, but we're now storing and sending all cookies that are set by the response headers.

Reply:
I guess it doesn't matter too much. I was asking because if it's referring to the API session, it would rely on the fact that the API _only_ sends the two cookies `JSESSION` and `SERVERID_SEARCH`. If it would send other cookies as well, these would just be "cookies", not "session_cookies". However it could still refer to the fact that all cookies can be seen as session information, so feel free to ignore that comment.